app/src/main/java/org/fossasia/phimpme/gallery/util/SecurityHelper.java

```diff
@@ -51,7 +51,7 @@ public void updateSecuritySetting(){
         String secur = SP.getString("SecuredLocalFolders", "");
         Gson gson = new Gson();
         String[] secfo = gson.fromJson(secur, String[].class);
-        if(secfo.length!=0){
+        if(secfo!=null){
```

Comment:
this condition only checks whether the string is not null but it doesn't check if it is empty, which means the body of the if condition can be executed even if the secfo string is empty. You can first check that the string is not null to avoid the crashing of the app and then check if the string is empty or not.

Reply:
secfo list will only have non-null value when securing local folders option is enabled, so there will one or more folders to secure thereby the secfo list will not be empty in that case so I guess only checking for secfo to be null will do.

Reply:
But what if you reset the settings, then all the values in the shared preference will be removed and it will return an empty string for secur which was in the case when you install the app.

Reply:
On resetting the settings, the secur string will store an empty string, which when passed to the gson.fromJson() function will return a null object, so it won't execute the statement in the if block and as the securedfolders array is only required when the security for local folders are enabled and as on resetting, the security is disabled so again just checking for null will do. Whenever the security for the local folders is enabled the secur string wouldn't  be empty and therefore passing it to the fromJson() function will return an array with path of the folders which are to be secured.